terraform/main.tf

```diff
@@ -2,69 +2,96 @@ locals {
   replicas = toset([for s in range(var.num_replicas) : tostring(s)])
 }
 
+module "fluenbit" {
+  source  = "./modules/monitoring/fluentbit"
+  image   = var.log_driver
+  network = docker_network.armonik.name
+}
 
-module "submitter" {
-  source         = "./modules/submitter"
-  container_name = "armonik.control.submitter"
-  core_tag       = var.core_tag
-  docker_image   = var.armonik_submitter_image
-  network        = docker_network.armonik_net.name
+module "seq" {
+  source  = "./modules/monitoring/seq"
+  image   = var.seq_image
+  network = docker_network.armonik.name
+}
 
-  implicit_dependencies = [
-  ]
+module "zipkin" {
+  source  = "./modules/monitoring/zipkin"
+  image   = var.zipkin_image
+  network = docker_network.armonik.name
+}
 
-  # handle fluentbit dependency as explicit, deployment fails otherwise
-  depends_on = [
-    docker_container.database,
-    docker_container.queue,
-    docker_container.object,
-    docker_container.zipkin,
-    docker_container.fluentbit
-  ]
+module "database" {
+  source  = "./modules/storage/mongo"
+  image   = var.database_image
+  network = docker_network.armonik.name
 }
 
-module "pollingagent" {
-  source          = "./modules/pollingAgent"
-  for_each        = local.replicas
-  container_name  = "armonik.compute.pollingagent${each.value}"
-  replica_counter = each.key
-  core_tag        = var.core_tag
-  docker_image    = var.armonik_pollingagent_image
-  network         = docker_network.armonik_net.name
-  socket_vol      = docker_volume.socket_vol[each.key].name
+module "object" {
+  source  = "./modules/storage/redis"
+  image   = var.object_image
+  network = docker_network.armonik.name
+}
+
+module "queue" {
+  source  = "./modules/queue/activemq"
+  image   = var.queue_image
+  network = docker_network.armonik.name
+}
 
-  implicit_dependencies = [
-  ]
+module "submitter" {
```

Comment:
I would pass the environment variables "generated" by the db and object storage as arguments. This would allow an implicit dependency, and make more modular.

Reply:
Agree, I'll work on that